File: service/src/main/java/by/rekuts/travelagency/service/impl/TourServiceImpl.java
package by.rekuts.travelagency.service.impl;

import by.rekuts.travelagency.domain.Tour;
import by.rekuts.travelagency.repository.TourRepository;
import by.rekuts.travelagency.repository.impl.CSVTourLoaderImpl;
import by.rekuts.travelagency.service.TourService;
import lombok.RequiredArgsConstructor;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@RequiredArgsConstructor
@Service
public class TourServiceImpl implements TourService {

    private final TourRepository tourRepository;
    private final CSVTourLoaderImpl csvTourLoaderImpl;

    @Transactional
    @Override
    public void insert(Tour tour) {
        tourRepository.save(tour);
    }

    @Transactional
    @Override
    public void update(Tour tour) {
        tourRepository.save(tour);
    }

    @Transactional
    @Override
    public void delete(int id) {
        tourRepository.deleteById(id);
    }

    @Transactional
    @Override
    public List<Tour> getList(Specification<Tour> specification) {
        return tourRepository.findAll(specification);
    }

    @Transactional
    @Override
    public void importCsvTours(String csvFileContent) {
        csvTourLoaderImpl.importCsvTours(csvFileContent);
    }
}

File: service/src/main/scala/by/rekuts/travelagency/service/impl/TourServiceImpl.scala
package by.rekuts.travelagency.service.impl

import java.util

import by.rekuts.travelagency.domain.Tour
import by.rekuts.travelagency.repository.{CSVTourLoader, Specification, TourRepository}
import by.rekuts.travelagency.service.TourService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class TourServiceImpl extends TourService {
  @Autowired
  private var tourRepository: TourRepository = _
  @Autowired
  private var csvTourLoader: CSVTourLoader = _

  @Transactional
  override def insert(tour: Tour): Unit = tourRepository.insert(tour)

  @Transactional
  override def update(tour: Tour): Unit = tourRepository.update(tour)

  @Transactional
  override def delete(id: Int): Unit = tourRepository.delete(id)

  @Transactional
  override def getList(specification: Specification[_]): util.List[Tour] = tourRepository.getList(specification)

  @Transactional
  override def importCsvTours(csvFileContent: String): Unit = csvTourLoader.importCsvTours(csvFileContent)
}
